### PR TITLE
Bug 1174645 - Improve job details message for missing logs

### PR DESCRIPTION
**Pull request by mandaltapesh**
Hi All,

Tested this on my system ..seems to work fine.. although feel free to test with more test cases/URLs.
your opinion will be valuable .

Regards,
Tapesh Mandal (IRC nick: tapesh)

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1012)

### CODE REVIEW

**Comment by tojonmz** [comment]
@mandaltapesh I haven't tested the branch, but can you shorten the commit message down to ~50 characters per general git [best practices](http://stackoverflow.com/questions/2290016/git-commit-messages-50-72-formatting).

You can also adjust the PR subject to be the same message. You'll want to use the a "present tense imperative" [style of wording](http://stackoverflow.com/questions/3580013/should-i-use-past-or-present-tense-in-git-commit-messages), which means the message will just state what the fix did.

**Comment by mandaltapesh** [comment]
Hi Jfrench,

Made the changes :)

Regards,
Tapesh Mandal

**Comment by tojonmz** [comment]
@mandaltapesh I'd suggest something like this, which will reduce it from 82 to 58 chars and in doing so eliminate the past tense "Added", and tell the user where the change occurred:

`Improve job details message for missing logs`

If you can use the same white space as you've done with your earlier contributions w.r.t. the commit message (between the "-") that would be good.

**Comment by mandaltapesh** [comment]
Hi jfrench,

i hope the above changes are satisfactory :)

Regards,
Tapesh Mandal

File: ui/plugins/controller.js [hunk 1]
```diff
@@ -157,7 +157,7 @@ treeherder.controller('PluginCtrl', [
                     // Provide a parse status for the model
                     var logsNotParsed = [];
                     $scope.jobLogsAllParsed = _.every($scope.job_log_urls, function(jlu) {
-                        if(jlu.parse_status === 'pending'){
+                        if(jlu.parse_status === 'pending' || jlu.parse_status === 'failed'){
```
Comment (tojonmz) [inline ui/plugins/controller.js]: While we're here let's put whitespace around the if statement
`if (condition) {`

File: ui/plugins/controller.js [hunk 2]
```diff
@@ -157,7 +157,7 @@ treeherder.controller('PluginCtrl', [
...
                             logsNotParsed.push(jlu);
                             return false;
                         }else{
```
Comment (tojonmz) [inline ui/plugins/controller.js]: And here.

File: ui/plugins/failure_summary/main.html [hunk 3]
```diff
@@ -86,7 +86,7 @@
             <span>is available. This panel will automatically recheck every 5 seconds.</span>
           </div>
         </li>
-        <li ng-if="!tabs.failureSummary.is_loading && !job_log_urls.length">
+        <li ng-if="!tabs.failureSummary.is_loading && !jobLogsAllParsed">
```
Comment (tojonmz) [inline ui/plugins/failure_summary/main.html]: Testing the PR locally I notice the Failure Summary appears blank on running, pending and coalesced jobs, where on production you would get a "No logs available for this job".

failuresummaryemptyrunning

**Comment by camd** [comment]
Hi Tapesh--  I was looking into this further today and while your branch works, I think this solution may not be quite ideal.  It would be better to show a unique message for log parsing failure.  Something like adding a line to the ``html`` file like:

    <li ng-if="!tabs.failureSummary.is_loading && logParseStatus === 'failed'">
        <div class="failure-summary-line-empty">
            <span>Log parsing failed.  Unable to generate failure summary</span>
        </div>
    </li>

You would also need to add a check of ``logParseStatus`` to the line that says "Log parsing complete" so that doesn't show when it is failed.

It would also be good to avoid the loop trying to load the "Bug Suggestions" artifact when the parsing has failed.  You could do so here:
https://github.com/mozilla/treeherder/blob/90ba77e596e2c3329e0b840fe45e661e8e2a0370/ui/plugins/failure_summary/controller.js#L68-68

Just add an additional check like: ``&& $scope.logParseStatus === "parsed"``

Then you won't see the screen reflashing over and over while it tries to fetch that artifact that will never come.

File: ui/plugins/controller.js (hunk 1 above)
Comment (camd) [inline ui/plugins/controller.js]: We can discard this change.  I apologize I initially confirmed this was the right path.  See my last comment on the overall PR.

**Comment by camd** [comment]
Thanks for all your work on this, Tapesh!  This will be a nice improvement when we've got it all squared away.  :)

**Comment by mandaltapesh** [comment]
Hi jfrench,

added whitespace wherever missing in both the controller files . Please test it. :)

Regards,
Tapesh Mandal

**Comment by mandaltapesh** [comment]
Hi camd,

Made modifications as you suggested. Made one more suitable change . :)
Regards,
Tapesh Mandal

**Comment by mandaltapesh** [comment]
made changes as directed by jfrench and camd :)

works fine with Ed's testcases :)

**Comment by camd** [comment]
Looks great!  Merging.

**Pull request merged**